core/commands/log_graph.py

```diff
@@ -314,7 +314,7 @@ def on_post_window_command(self, window, command_name, args):
 def colorize_dots(view):
     # type: (sublime.View) -> None
     dots = tuple(_find_dots(view))
-    _colorize_dots(view.id(), dots, view.change_count())
```

Comment:
If the buffer changed I think we should colorize over againg

Reply:
Look at hash function of `Char` which does this check for us. 👍

Reply:
> Since a dictionary is used to cache results, the positional and keyword arguments to the function must be hashable.
https://docs.python.org/3/library/functools.html

Does this mean that is calles __hash__?

Reply:
Yes!